src/devices/src/virtio/net/device.rs

```diff
@@ -722,6 +735,76 @@ impl VirtioDevice for Net {
     }
 }
 
+#[derive(Versionize)]
+pub struct NetState {
+    id: String,
+    tap_if_name: String,
+    rx_rate_limiter_state: RateLimiterState,
+    tx_rate_limiter_state: RateLimiterState,
+    allow_mmds_requests: bool,
+    guest_mac: Option<Vec<u8>>,
```

Comment:
If we want to be future looking about how we structure and name states, I would move guest_mac to a new struct NetConfigSpace and include it as config_space in the NetState. As we add features to any device in the future we will need to extend the config space with other fields that can be read/written by the guest. 

@andreeaflorescu might have some input her also.

Reply:
Would it make sense to also keep `allow_mmds_requests` in the NetConfigSpace?

Reply:
That structure is actually treated as a buffer (`impl ByteValued for NetConfigSpace`) and read/written by the guest virtio net driver.

Only guest mac needs to be there for now. Other virtio features (for example multiqueue) will require us to add more fields in this structure.